yamux/src/stream.rs

```diff
@@ -308,25 +301,25 @@ impl StreamHandle {
         Ok(())
     }
 
-    fn check_self_state(&mut self) -> Result<(), io::Error> {
+    // Returns `Ok(Some(..))` if someting out of thin air should be returned to the user.
+    // Otherwise, returns `Err(Error)` whenever unable to read bytes.
+    fn check_self_state(&mut self) -> io::Result<Option<io::Result<()>>> {
```

Comment:
```suggestion
    fn check_self_state(&mut self) -> io::Result<bool> {
```
I think it's not a good return type, why should `fn check_self_state()` care about what's the return value will be used?

Reply:
agree. updated.